pkg/operator/k8sutil/replicaset.go

```diff
@@ -24,8 +24,7 @@ import (
 )
 
 // DeleteReplicaSet makes a best effort at deleting a deployment and its pods, then waits for them to be deleted
-func DeleteReplicaSet(clientset kubernetes.Interface, namespace, name string) error {
-	ctx := context.TODO()
+func DeleteReplicaSet(ctx context.Context, clientset kubernetes.Interface, namespace, name string) error {
```

Comment:
Interesting that `DeleteReplicaSet` is not used anywhere, it seems useful though, so let's keep it.